### DISCUSSION TITLE
Spacy ner scorer return all zeros when using dropout

### DISCUSSION THREAD

**Discussion by mjvallone** [explosion/spaCy#7775] [Help: Coding & Implementations]
Hi all,

We are trying to get scores on our training and validation set using Scorer and after a couple of epochs it returns zero for every possible score (f_score, recall, precision and scores per entity type - but NER losses decreases) when **using non-zero dropout**.
If we use zero dropout, it returns non-zero scores as expected.
On our training, we are reducing dropout rate on every epoch, this is the config callback to do that: 
`{"f": "change_dropout_fixed", "step": -0.01, "until": 0.3},`

(Not so) related info: we are using a couple of callbacks to apply configuration settings to our model training.

Here is our code:

```
def change_dropout_fixed(step=0.01, until=0.5):
  """
  [experimental] change the dropout each epoch
  :param step amount to change per epoch
  :param until limit for dropout change  
  """
  def change_dropout_fixed_cb(state, logger, model, optimizer, disabled_pipes):

    if step > 0 and state["dropout"] < until:
      state["dropout"] += step
      logger.info(f"[change_dropout_fixed] touching dropout. New value {state['dropout']}")
    elif step < 0 and state["dropout"] > until:
      # negative step
      state["dropout"] += step
      logger.info(f"[change_dropout_fixed] touching dropout. New value {state['dropout']}")

    else:
      logger.info(f"[change_dropout_fixed] No more room for touching dropout")

    return state

  return change_dropout_fixed_cb

```

Our configuration data (only what matters):
```
  "train_config": {
    "path_data_training": "",
    "path_data_validation": "",
    "is_raw": false,
    "train_subset": 20,
    "model_path": "",
    "entities": "PER, LOC, DIRECCIÓN, ARTÍCULO",
    "epochs": 10,
    "optimizer": {"learn_rate": 0.0004},
    "dropout": 0.5,
    "batch_size": 4,
    "callbacks": {
      "on_iteration":[
        {"f": "print_scores_on_epoch"},
        {"f": "early_stop", "epochs": 20, "diff": 0.1},
        {"f": "reduce_lr_on_plateau", "epochs": 4, "diff": 1, "step": 0.001},
        {"f": "sleep", "secs": 2, "log": true},
        {"f": "save_best_model", "path_best_model": "models/current", "threshold": 1},
        {"f": "update_best_scores"},
        {"f": "sleep", "secs": 1, "log": true}
      ],
      "on_batch":[],
      "on_stop":[
        {"f": "log_best_scores"},
        {"f": "save_csv_history"}
      ]
    }    
  },
```
We've also tried with fixed dropout rate (for every epoch) and something similar happens.
I'ts weird that  scores decreases on every epoch until reaches zero.

Here is the training log:

epoch | batches | lr | dropout | ner | f_score | recall | precision | per_type_score
-- | -- | -- | -- | -- | -- | -- | -- | --
1 | 5 | 0.0004 | 0.5 | 101754.320159912 | 31.4210441216909 | 29.3743312805021 | 36.4487642633206 | {'ARTÍCULO': 29.688508623992494, 'LOC': 20.76388888888889, 'DIRECCIÓN': 25.20310391363023, 'PER': 23.92490842490842}
2 | 5 | 0.0004 | 0.5 | 98618.123260498 | 27.9841887991107 | 25.4461746183882 | 33.6707124592837 | {'ARTÍCULO': 27.33616571496354, 'LOC': 17.083333333333332, 'DIRECCIÓN': 20.736263736263737, 'PER': 18.857054504421995}
3 | 5 | 0.0004 | 0.5 | 95440.654586792 | 26.140183942938 | 23.3058331550451 | 31.973405321248 | {'ARTÍCULO': 26.665122872019424, 'LOC': 7.857142857142857, 'DIRECCIÓN': 16.537626116573485, 'PER': 18.55704480704481}
4 | 5 | 0.0004 | 0.5 | 92490.5054626465 | 12.4752777526294 | 8.59469570189384 | 27.3175465838509 | {'ARTÍCULO': 12.181151544135764, 'LOC': 0.0, 'DIRECCIÓN': 6.321195144724557, 'PER': 11.584763750461425}
5 | 5 | 0.0004 | 0.5 | 90090.1886444092 | 5.95134930849217 | 3.55688040982159 | 20.9848484848485 | {'ARTÍCULO': 4.787878787878788, 'LOC': 0.0, 'DIRECCIÓN': 0.0, 'PER': 7.326749826749827}
6 | 5 | 0.0004 | 0.5 | 85701.8920898438 | 0 | 0 | 0 | {'ARTÍCULO': 0.0, 'LOC': 0.0, 'DIRECCIÓN': 0.0, 'PER': 0.0}
7 | 5 | 0.0004 | 0.5 | 82723.9035186768 | 0 | 0 | 0 | {'ARTÍCULO': 0.0, 'LOC': 0.0, 'DIRECCIÓN': 0.0, 'PER': 0.0}
8 | 5 | 0.0004 | 0.5 | 79870.3180389404 | 0 | 0 | 0 | {'ARTÍCULO': 0.0, 'LOC': 0.0, 'DIRECCIÓN': 0.0, 'PER': 0.0}
9 | 5 | 0.0004 | 0.5 | 76869.5482940674 | 0 | 0 | 0 | {'ARTÍCULO': 0.0, 'LOC': 0.0, 'DIRECCIÓN': 0.0, 'PER': 0.0}
10 | 5 | 0.0004 | 0.5 | 74832.4391326904 | 0 | 0 | 0 | {'ARTÍCULO': 0.0, 'LOC': 0.0, 'DIRECCIÓN': 0.0, 'PER': 0.0}

I hope someone passed through this because it's kindof blocking us this issue.

## Your Environment
* Operating System: Ubuntu 20.04.2 LTS
* Python Version Used: 3.6.1
* spaCy Version Used: 2.3.2
* Environment Information: -

**Comment by svlandeg** [comment]
Hi! As a maintenance note: this is a good question for the discussions forum, so I'll move it there. The original issue will be closed and will link through to the open discussion thread.
